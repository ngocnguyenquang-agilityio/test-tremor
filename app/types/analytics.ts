export interface AnalyticsInfoData {
  id: string;
  photo: string;
  name: string;
  description: string;
  price: number;
  location: string;
}

export interface LineChartData {
  date: string;
  "Desktop apps"?: number;
  "Mobile apps"?: number;
}

export interface LineChart {
  id: string;
  display: string;
  descValue: string;
  desc: string;
  modified: string;
  data: LineChartData[];
}

type TFixed = {
  eventType: "dot" | "category" | "bar" | "slice" | "bubble";
  categoryClicked: string;
};

type TBaseEvent = TFixed & {
  [key: string]: number | string;
};

export type TEvent = TBaseEvent | null | undefined;
