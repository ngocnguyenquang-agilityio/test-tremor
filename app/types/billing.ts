import { AGGREGATION_TYPE } from "@/constants";

export interface Invoice {
  id: number;
  createdAt: string;
  invoicePrefix: string;
  totalCost: number;
}

export interface Transaction {
  id: number;
  createdAt: string;
  service: string;
  amount: number;
  type: number;
  status: number;
}
export interface BillingCardData {
  id: number;
  type: string;
  expire: string;
  cardNumber: string;
  holderFullName: string;
  cardLast4Digit: string;
}

export interface SalaryCardData {
  value: number;
  type: AGGREGATION_TYPE;
  currency?: string;
}
