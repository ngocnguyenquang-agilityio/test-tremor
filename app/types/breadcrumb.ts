export interface BreadcrumbLink {
  name: string;
  url: string;
}
