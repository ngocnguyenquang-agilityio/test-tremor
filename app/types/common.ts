export interface Step {
  index: number;
  content: string;
}

export interface IMedia {
  image: string;
}

export interface CdnResponse {
  data: {
    width: number;
    height: number;
    image: ImageInfo;
  };
}

export interface ImageInfo {
  filename: string;
  url: string;
}

export interface SortItem {
  key: string;
  direction: string;
}

export type TSearchParams = {
  query?: string;
  filter?: string;
  page?: number;
  sortBy?: string;
};
