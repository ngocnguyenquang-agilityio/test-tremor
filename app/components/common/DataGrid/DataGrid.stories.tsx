// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import DataGrid from ".";

// Types
import { ColumnType, TOrder } from "@/types";

// Constants
import { STATUS_TEXT } from "@/constants";

const meta = {
  title: "Components/common/DataGrid",
  tags: ["autodocs"],
  component: DataGrid,
} as Meta<typeof DataGrid>;

export default meta;

type Story = StoryObj<typeof meta>;

const data: TOrder[] = [
  {
    id: 10425,
    created_at: "2023-11-01T10:20:00+00:00",
    status: 1,
    customer: {
      id: 15,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      full_name: "Orlando Imieto",
    },
    revenue: 44.9,
    products: [
      {
        id: 6,
        name: "Phone Case Pink",
        count: 2,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 90,
      },
    ],
  },
  {
    id: 12345,
    created_at: "2023-11-01T03:20:00+00:00",
    status: 0,
    customer: {
      id: 1,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      full_name: "Orlando Imieto",
    },
    revenue: 140.2,
    products: [
      {
        id: 12,
        name: "Office Papers",
        count: 1,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 77,
      },
    ],
  },
  {
    id: 12350,
    created_at: "2023-11-01T11:03:00+00:00",
    status: 0,
    customer: {
      id: 9,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      full_name: "Orlando Imieto",
    },
    revenue: 112.5,
    products: [
      {
        id: 10,
        name: "Watter Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 7,
      },
    ],
  },
];

const columns: ColumnType<TOrder>[] = [
  {
    key: "id",
    title: "Id",
  },
  {
    key: "created_at",
    title: "Date",
  },
  {
    key: "status",
    title: "Status",
    customNode: (_, { status }) => (
      <div className="flex justify-start text-xs dark:text-white font-semibold leading-[15px] tracking-[0.4px] capitalize order-status">
        <p className="text-xs dark:text-white font-semibold leading-[15px] tracking-[0.4px] capitalize order-status">
          {STATUS_TEXT[status]}
        </p>
      </div>
    ),
  },
];

export const DataTableDefault: Story = {
  render: () => <DataGrid data={data} columns={columns} />,
};
