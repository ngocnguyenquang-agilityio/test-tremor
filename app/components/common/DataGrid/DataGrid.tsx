"use client";

import { Card, Table } from "@tremor/react";

// Components
import {
  Pagination,
  DataGridHeader,
  DataGridBody,
  LoadingIndicator,
} from "@/components";

// Types
import { ColumnType } from "@/types";

// Hooks
import { useEffect, useMemo, useState } from "react";

// Helpers
import { useSortableTable } from "@/hooks/useSortableTable";

interface DataTableProps<T> {
  data: T[];
  columns: ColumnType<T>[];
  pageSize?: number;
  filterBy: string;
  keyword: string;
  className?: string;
  hasPagination?: boolean;
  currentPageNumber?: number;
  total?: number;
  onPageChange?: (page: number) => void;
}

const DataGrid = <T,>({
  data,
  columns,
  pageSize = 10,
  filterBy,
  keyword,
  className = "",
  hasPagination = true,
  total,
  currentPageNumber,
  onPageChange,
}: DataTableProps<T>) => {
  const [currentPage, setCurrentPage] = useState(1);
  const [loading, setLoading] = useState(false);

  const [tableData, handleSorting] = useSortableTable<T>(data);

  const currentTableData = useMemo(() => {
    const firstPageIndex = (currentPage - 1) * pageSize;
    const lastPageIndex = firstPageIndex + pageSize;

    // Check if tableData is an array before slicing
    let dataSorted: T[] = [];
    if (Array.isArray(tableData)) {
      dataSorted = tableData.slice(firstPageIndex, lastPageIndex);
    }

    return dataSorted;
  }, [currentPage, tableData, pageSize]);

  useEffect(() => {
    const column = columns.find(item => item.sortbyOrder);

    if (column) {
      handleSorting(column.key, column.sortbyOrder);
    }
  }, [columns, handleSorting]);

  useEffect(() => {
    setLoading(true);

    // Delay to check show loading
    setTimeout(() => {
      setLoading(false);
    }, 100);
  }, [filterBy, keyword, currentPage, pageSize]);

  if (loading) {
    return (
      <LoadingIndicator
        additionalClass="min-h-[500px] flex justify-center items-center"
        width={8}
        height={8}
        isFullWidth={false}
        fillColor="river-bed-500"
      />
    );
  }

  return (
    <Card
      className={`p-0 border-none ring-0 dark:bg-dark-tremor-primary overflow-x-auto ${className}`}>
      <div className="flex flex-col items-start justify-start my-2">
        <Table className="w-full">
          <DataGridHeader
            columns={columns}
            handleSorting={
              handleSorting as (sortField: string, sortOrder: string) => void
            }
          />
          <DataGridBody columns={columns} data={currentTableData} />
        </Table>
        {hasPagination && (
          <Pagination
            currentPage={currentPageNumber!}
            pageSize={pageSize}
            totalCount={total!}
            onPageChange={onPageChange!}
          />
        )}
      </div>
    </Card>
  );
};

export default DataGrid;
