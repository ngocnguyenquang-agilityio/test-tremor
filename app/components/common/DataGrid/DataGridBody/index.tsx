// Components
import { TableCell, TableRow } from "@tremor/react";

// Types
import { ColumnType } from "@/types";
import { getObjectValue } from "@/helpers";

// Hooks
import { useId } from "react";

interface DataTableBodyProps<T> {
  data: T[];
  columns: ColumnType<T>[];
}

const DataGridBody = <T,>({ data, columns }: DataTableBodyProps<T>) => {
  const rowId = useId();

  return (
    <>
      {data.map(item => (
        <TableRow key={`table-body-${rowId}`}>
          {columns.map(column => (
            <TableCell
              key={`table-row-cell-${column.key}`}
              className="px-6 py-5 border-0 border-b border-gray-100">
              {column.customNode ? (
                column.customNode(column, item)
              ) : (
                <p className="text-xs font-semibold">
                  {getObjectValue(item, column.key as keyof T) as string}
                </p>
              )}
            </TableCell>
          ))}
        </TableRow>
      ))}
    </>
  );
};

export default DataGridBody;
