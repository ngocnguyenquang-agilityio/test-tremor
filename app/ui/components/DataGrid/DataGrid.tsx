"use client";

import { Card, Table, TableBody, TableCell, TableRow } from "@tremor/react";
import { usePathname, useRouter, useSearchParams } from "next/navigation";

// Components

import {
  DataGridBody,
  DataGridHeader,
  LoadingIndicator,
  Pagination,
} from "@/ui/components";

// Types
import { ColumnType } from "@/types";

// Hooks
import { useEffect, useMemo, useState, useTransition } from "react";

// Helpers
import { useSortableTable } from "@/hooks/useSortableTable";

interface DataTableProps<T> {
  data: T[];
  columns: ColumnType<T>[];
  pageSize?: number;
  className?: string;
  hasPagination?: boolean;
  currentPageNumber?: number;
  total?: number;
  defaultCurrentPage?: number;
  disableLoading?: boolean;
}

const DataGrid = <T,>({
  data,
  columns,
  pageSize = 10,
  className = "",
  hasPagination = true,
  total,
  currentPageNumber,
  defaultCurrentPage = 1,
  disableLoading = false,
}: DataTableProps<T>) => {
  const searchParams = useSearchParams();
  const pathname = usePathname();
  const { replace } = useRouter();
  const params = new URLSearchParams(searchParams);

  const [currentPage, setCurrentPage] = useState(defaultCurrentPage);

  const [loading, setLoading] = useState(false);

  const [tableData, handleSorting] = useSortableTable<T>(data);

  let [isPending, startTransition] = useTransition();

  useEffect(() => {
    if (isPending) return;

    // THIS CODE WILL RUN AFTER THE SERVER ACTION
    setLoading(false);
  }, [isPending]);

  // Handle page in pagination changed
  const handlePageChange = (page: number) => {
    startTransition(() => {
      setLoading(true);
    });

    if (page === 1) {
      params.delete("page");
    } else {
      params.set("page", page.toString());
    }
    const newFullPathname = `${pathname}?${params.toString()}`;
    replace(newFullPathname);
    setCurrentPage(page);
  };

  const currentTableData = useMemo(() => {
    const firstPageIndex = (currentPage - 1) * pageSize;
    const lastPageIndex = firstPageIndex + pageSize;

    // Check if tableData is an array before slicing
    let dataSorted: T[] = [];
    if (Array.isArray(tableData)) {
      dataSorted = tableData.slice(firstPageIndex, lastPageIndex);
    }

    return dataSorted;
  }, [currentPage, tableData, pageSize]);

  return (
    <Card
      className={`p-0 border-none ring-0 dark:bg-dark-tremor-primary overflow-x-auto ${className}`}>
      <div className="flex flex-col items-start justify-start my-2">
        <Table className="w-full">
          <DataGridHeader
            columns={columns}
            handleSorting={
              handleSorting as (sortField: string, sortOrder: string) => void
            }
          />
          {(loading || isPending) && !disableLoading ? (
            <TableBody>
              <TableRow>
                <TableCell colSpan={6}>
                  <LoadingIndicator
                    additionalClass="min-h-[500px] flex justify-center items-center"
                    width={8}
                    height={8}
                    isFullWidth={false}
                    fillColor="river-bed-500"
                  />
                </TableCell>
              </TableRow>
            </TableBody>
          ) : (
            <DataGridBody columns={columns} data={currentTableData} />
          )}
        </Table>
        {hasPagination && (
          <Pagination
            currentPage={currentPageNumber!}
            pageSize={pageSize}
            totalCount={total!}
            onPageChange={handlePageChange}
          />
        )}
      </div>
    </Card>
  );
};

export default DataGrid;
