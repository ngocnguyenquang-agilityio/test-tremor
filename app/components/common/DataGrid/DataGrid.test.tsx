import { render } from "@testing-library/react";

//Components
import DataGrid from "./DataGrid";

// Types
import { TOrder, ColumnType } from "@/types";

describe("Data Table Testing", () => {
  const data: TOrder[] = [
    {
      id: 10425,
      created_at: "2023-11-01T10:20:00+00:00",
      status: 1,
      customer: {
        id: 15,
        avatar:
          "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
        full_name: "Orlando Imieto",
      },
      revenue: 44.9,
      products: [
        {
          id: 6,
          name: "Phone Case Pink",
          count: 2,
          url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
          price: 90,
        },
      ],
    },
    {
      id: 12345,
      created_at: "2023-11-01T03:20:00+00:00",
      status: 0,
      customer: {
        id: 1,
        avatar:
          "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
        full_name: "Orlando Imieto",
      },
      revenue: 140.2,
      products: [
        {
          id: 12,
          name: "Office Papers",
          count: 1,
          url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
          price: 77,
        },
      ],
    },
    {
      id: 12350,
      created_at: "2023-11-01T11:03:00+00:00",
      status: 0,
      customer: {
        id: 9,
        avatar:
          "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
        full_name: "Orlando Imieto",
      },
      revenue: 112.5,
      products: [
        {
          id: 10,
          name: "Watter Bottle India",
          count: 3,
          url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
          price: 7,
        },
      ],
    },
  ];

  const columns: ColumnType<TOrder>[] = [
    {
      key: "id",
      title: "Id",
    },
    {
      key: "created_at",
      title: "Date",
    },
  ];

  it("should match snapshot", () => {
    const { container } = render(<DataGrid data={data} columns={columns} />);
    expect(container).toMatchSnapshot();
  });
});
