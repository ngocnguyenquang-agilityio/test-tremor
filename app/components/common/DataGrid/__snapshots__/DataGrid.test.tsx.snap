// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Testing should match snapshot 1`] = `
<div>
  <div
    class="tremor-Card-root relative w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-0 border-none ring-0 dark:bg-dark-tremor-primary overflow-x-auto"
  >
    <div
      class="flex flex-col items-start justify-start my-2"
    >
      <div
        class="tremor-Table-root overflow-auto w-full"
      >
        <table
          class="tremor-Table-table w-full tabular-nums text-tremor-default text-tremor-content dark:text-dark-tremor-content"
        >
          <thead
            class="tremor-TableHead-root text-left text-tremor-content dark:text-dark-tremor-content"
          >
            <tr
              class="tremor-TableRow-row"
            >
              <th
                class="tremor-TableHeaderCell-root sticky whitespace-nowrap text-left text-tremor-content top-0 px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold opacity-70 uppercase"
              >
                <div
                  class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                >
                  <div
                    class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                  >
                    Id
                  </div>
                  <div
                    class="tremor-Flex-root flex w-full justify-between items-center relative flex-col mx-4"
                  >
                    <div
                      class="tremor-Flex-root flex w-full flex-row items-center absotule top-0 justify-end"
                    >
                      <button
                        class="tremor-Button-root flex-shrink-0 inline-flex group font-medium outline-none text-sm text-tremor-brand dark:text-dark-tremor-brand bg-transparent hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis w-1.5 h-1.5 justify-center items-center opacity-50 dark:opacity-100"
                      >
                        <span
                          class="tremor-Button-text text-sm whitespace-nowrap"
                        >
                          <svg
                            class="w-4 h-4"
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M7 14l5-5 5 5z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                    <div
                      class="tremor-Flex-root flex w-full flex-row items-center absotule top-0 justify-end"
                    >
                      <button
                        class="tremor-Button-root flex-shrink-0 inline-flex group font-medium outline-none text-sm text-tremor-brand dark:text-dark-tremor-brand bg-transparent hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis w-1.5 h-1.5 justify-center items-center opacity-50 dark:opacity-100"
                      >
                        <span
                          class="tremor-Button-text text-sm whitespace-nowrap"
                        >
                          <svg
                            class="w-4 h-4"
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="tremor-TableHeaderCell-root sticky whitespace-nowrap text-left text-tremor-content top-0 px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold opacity-70 uppercase"
              >
                <div
                  class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                >
                  <div
                    class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                  >
                    Date
                  </div>
                  <div
                    class="tremor-Flex-root flex w-full justify-between items-center relative flex-col mx-4"
                  >
                    <div
                      class="tremor-Flex-root flex w-full flex-row items-center absotule top-0 justify-end"
                    >
                      <button
                        class="tremor-Button-root flex-shrink-0 inline-flex group font-medium outline-none text-sm text-tremor-brand dark:text-dark-tremor-brand bg-transparent hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis w-1.5 h-1.5 justify-center items-center opacity-50 dark:opacity-100"
                      >
                        <span
                          class="tremor-Button-text text-sm whitespace-nowrap"
                        >
                          <svg
                            class="w-4 h-4"
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M7 14l5-5 5 5z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                    <div
                      class="tremor-Flex-root flex w-full flex-row items-center absotule top-0 justify-end"
                    >
                      <button
                        class="tremor-Button-root flex-shrink-0 inline-flex group font-medium outline-none text-sm text-tremor-brand dark:text-dark-tremor-brand bg-transparent hover:text-tremor-brand-emphasis dark:hover:text-dark-tremor-brand-emphasis w-1.5 h-1.5 justify-center items-center opacity-50 dark:opacity-100"
                      >
                        <span
                          class="tremor-Button-text text-sm whitespace-nowrap"
                        >
                          <svg
                            class="w-4 h-4"
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="tremor-TableBody-root align-top overflow-x-auto divide-y divide-tremor-border dark:divide-dark-tremor-border"
          >
            <tr
              class="tremor-TableRow-row"
            >
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap tabular-nums text-left p-4 px-6 py-5 border-0 border-b border-gray-100"
              >
                <p
                  class="text-xs font-semibold"
                >
                  10425
                </p>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap tabular-nums text-left p-4 px-6 py-5 border-0 border-b border-gray-100"
              >
                <p
                  class="text-xs font-semibold"
                >
                  2023-11-01T10:20:00+00:00
                </p>
              </td>
            </tr>
            <tr
              class="tremor-TableRow-row"
            >
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap tabular-nums text-left p-4 px-6 py-5 border-0 border-b border-gray-100"
              >
                <p
                  class="text-xs font-semibold"
                >
                  12345
                </p>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap tabular-nums text-left p-4 px-6 py-5 border-0 border-b border-gray-100"
              >
                <p
                  class="text-xs font-semibold"
                >
                  2023-11-01T03:20:00+00:00
                </p>
              </td>
            </tr>
            <tr
              class="tremor-TableRow-row"
            >
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap tabular-nums text-left p-4 px-6 py-5 border-0 border-b border-gray-100"
              >
                <p
                  class="text-xs font-semibold"
                >
                  12350
                </p>
              </td>
              <td
                class="tremor-TableCell-root align-middle whitespace-nowrap tabular-nums text-left p-4 px-6 py-5 border-0 border-b border-gray-100"
              >
                <p
                  class="text-xs font-semibold"
                >
                  2023-11-01T11:03:00+00:00
                </p>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
